packages/flutter/lib/src/painting/box_border.dart

```diff
@@ -595,8 +625,9 @@ class Border extends BoxBorder {
     }());
     assert(() {
       if (shape != BoxShape.rectangle) {
```

Comment:
This should probably be a switch statement instead of an "if", just in case we add more shapes in the future.

Reply:
It got slightly weird (wish you could return a switch :( hopefully soon), but it is there, I agree with the more shapes in future.